### PR TITLE
telegram-desktop (and forks): make webkitgtk optional, and update to latest version

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      libpulseaudio
+      hunspell
+      jemalloc
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_15
+      libicns
+    ];
+
+  dontWrapQtApps = true;
```
Comment (ilya-fedin) [inline pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix:114]: This does nothing as wrapQtAppsHook is not in nativeBuildInputs

Comment (NickCao) [reply pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix:114]: If it's not set qtbase's setup hook would complain.

Comment (ilya-fedin) [reply pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix:114]: ah, ok

**Comment by ilya-fedin** [comment]
Can you also remove kotato's with-webkit.nix and the corresponding entry from all-packages.nix?

**Comment by NickCao** [comment]
> Can you also remove kotato's with-webkit.nix and the corresponding entry from all-packages.nix?

Done. materialgram is still broken, maybe also switch it to qt5?

**Comment by ilya-fedin** [comment]
> Done. materialgram is still broken, maybe also switch it to qt5?

Maybe... Another way is to switch tdesktop to ffmpeg_6 which should make tdesktop, materialgram and 64gram based on 5.7x build.

**Comment by NickCao** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353524`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>_64gram</li>
    <li>kotatogram-desktop</li>
    <li>materialgram</li>
    <li>telegram-desktop</li>
  </ul>
</details>

**Comment by ilya-fedin** [comment]
hmm :thinking: 
```
nix-build --expr 'with import ./. { }; kotatogram-desktop.override { withWebkit = false; }'
error:
       … while calling a functor (an attribute set with a '__functor' attribute)

         at «string»:1:22:

            1| with import ./. { }; kotatogram-desktop.override { withWebkit = false; }
             |                      ^

       … while calling a functor (an attribute set with a '__functor' attribute)

         at /home/ilya/nixpkgs/lib/customisation.nix:144:43:

          143|       # Re-call the function but with different arguments
          144|       overrideArgs = mirrorArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                           ^
          145|       # Change the result of the function call by applying g to it

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: function 'anonymous lambda' called with unexpected argument 'withWebkit'

       at /home/ilya/nixpkgs/pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:1:1:

            1| {
             | ^
            2|   lib,
```

**Comment by NickCao** [comment]
Fixed, in a ugly way...

**Review by ilya-fedin — APPROVED** [review]

**Comment by ilya-fedin** [comment]
meh, ofborg seem to not to try to build on darwin (according to https://logs.ofborg.org/?key=nixos/nixpkgs.353524) :(

**Review by JohnRTitor — APPROVED** [review]
```
johnrtitor@darwin01 ~ % nix build "github:nixos/nixpkgs/refs/pull/353524/merge#_64gram" "github:nixos/nixpkgs/refs/pull/353524/merge#kotatogram-desktop" "github:nixos/nixpkgs/refs/pull/353524/merge#materialgram" "github:nixos/nixpkgs/refs/pull/353524/merge#telegram-desktop" --print-out-paths 

/nix/store/b5h15jgzq6hkqwcdajls8ycnb19jh7wl-64gram-1.1.45
/nix/store/24v6afl6i3ay1v70gd1fvyzsh0ss5mnq-kotatogram-desktop-1.4.9-unstable-2024-09-27
/nix/store/hws4kxa355cdwklvriz3mqnc47ala9na-materialgram-5.7.0.1
/nix/store/90rk4m1v7bpgajvnw0mb8s4n1n74xq0s-telegram-desktop-5.7.1

nix build "github:nixos/nixpkgs/refs/pull/353524/merge#_64gram"      5.78s user 15.87s system 0% cpu 1:32:28.51 total
```

**Pull request merged**